### PR TITLE
Cleaning up integration tests

### PR DESCRIPTION
**Pull request by slevenick**
Cleanup InSpec integration tests

-----------------------------------------------------------------
# [all]

### CODE REVIEW

File: .ci/acceptance-tests/inspec-integration.sh [hunk 1]
```diff
@@ -17,16 +17,27 @@ export GOPATH=${PWD}/go
...
-git clone https://github.com/slevenick/inspec-gcp.git
+export CLOUD_SDK_REPO="cloud-sdk-stretch"
+echo "deb http://packages.cloud.google.com/apt $CLOUD_SDK_REPO main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+apt-get update && apt-get install google-cloud-sdk -y
 
-# new train plugin not published yet, install locally for now
-pushd inspec-gcp
-bundle
+gcloud auth activate-service-account terraform@graphite-test-sam-chef.iam.gserviceaccount.com --key-file=$GOOGLE_CLOUD_KEYFILE_JSON
```
Comment (rambleraptor) [inline .ci/acceptance-tests/inspec-integration.sh]: Do we have to use gcloud to get the service account working? Seems like a lot of overhead

File: .ci/acceptance-tests/inspec-integration.sh [hunk 2]
```diff
@@ -60,7 +71,7 @@ bundle
 # Service accounts take several minutes to be authorized everywhere
 set +e
 
-for i in {1..50}
+for i in {1..30}
```
Comment (rambleraptor) [inline .ci/acceptance-tests/inspec-integration.sh]: What's the for loop for?

File: provider/inspec.rb [hunk 3]
```diff
@@ -56,6 +56,60 @@ def generate_resource(data)
           File.join(target_folder, "google_#{data[:product_name]}_#{name}".pluralize + '.rb')
       )
       generate_documentation(data)
+      generate_properties(data)
+    end
+
+    def generate_properties(data)
```
Comment (rambleraptor) [inline provider/inspec.rb]: What's going on with all of this?

File: .ci/acceptance-tests/inspec-integration.sh (hunk 2 above)
Comment (slevenick) [reply .ci/acceptance-tests/inspec-integration.sh]: Newly created service accounts are not reliably recognized by authentication servers immediately. So we keep trying until it works or has ran for enough time to say that something other than authentication is the issue. This is not ideal, but I don't see another option at this point.

File: provider/inspec.rb (hunk 3 above)
Comment (slevenick) [reply provider/inspec.rb]: InSpec was using the chef/puppet way of generating properties & nested object files. This is reproducing that as it was removed.

File: .ci/acceptance-tests/inspec-integration.sh (hunk 1 above)
Comment (slevenick) [reply .ci/acceptance-tests/inspec-integration.sh]: gcloud/gsutils is using the provided service account to pull down the contents of a private bucket that contain my train plugin until it is published. Planning on also storing VCR cassettes there

File: .ci/acceptance-tests/inspec-integration.sh (hunk 2 above)
Comment (SirGitsalot) [reply .ci/acceptance-tests/inspec-integration.sh]: We've had some offline discussions about alternatives, and they're all terrible. This should be OK until we come up with something better.

**Pull request merged**